src/config/parse.c

```diff
@@ -873,302 +971,491 @@ configParse(unsigned int argListSize, const char *argList[], bool resetLogLevel)
-                                {
-                                    THROW_FMT(
-                                        OptionInvalidValueError, "'%s' cannot contain // for '%s' option", strZ(value),
-                                        cfgOptionName(optionId));
+                                        configOptionValue->value = varNewStrZ(value);
+                                    }
                                 }
-
-                                // If the path ends with a / we'll strip it off (unless the value is just /)
-                                if (strEndsWithZ(value, "/") && strSize(value) != 1)
-                                    strTrunc(value, (int)strSize(value) - 1);
+                                MEM_CONTEXT_END();
                             }
-
-                            // If the option has an allow list then check it
-                            if (cfgDefOptionAllowList(commandId, optionDefId) &&
-                                !cfgDefOptionAllowListValueValid(commandId, optionDefId, strZ(value)))
+                            // Else error if option is required and help was not requested
+                            else if (cfgDefOptionRequired(config->command, optionId) && !config->help)
                             {
+                                const char *hint = "";
+
+                                if (cfgDefOptionSection(optionId) == cfgDefSectionStanza)
+                                    hint = "\nHINT: does this stanza exist?";
+
                                 THROW_FMT(
-                                    OptionInvalidValueError, "'%s' is not allowed for '%s' option", strZ(value),
-                                    cfgOptionName(optionId));
+                                    OptionRequiredError, "%s command requires option: %s%s", cfgCommandName(config->command),
+                                    cfgOptionKeyIdxName(optionId, optionKeyIdx), hint);
                             }
-
-                            cfgOptionSet(optionId, parseOption->source, VARSTR(value));
                         }
                     }
-                    else if (parseOption->negate)
-                        cfgOptionSet(optionId, parseOption->source, NULL);
-                    // Else try to set a default
-                    else
-                    {
-                        // Get the default value for this option
-                        const char *value = cfgDefOptionDefault(commandId, optionDefId);
+                }
+            }
+        }
 
-                        if (value != NULL)
-                            cfgOptionSet(optionId, cfgSourceDefault, VARSTRZ(value));
-                        else if (cfgOptionIndex(optionId) == 0 && cfgDefOptionRequired(commandId, optionDefId) &&
-                                 !cfgCommandHelp())
-                        {
-                            const char *hint = "";
+        // Initialize config
+        cfgInit(config);
```

Comment:
Why are we not calling `cfgInit(config)` AFTER the `// Set option group default index.` below? Seems odd to set everything except `config->optionGroup[groupId].indexDefault` beforehand.

Reply:
For this particular piece of code it's useful to be able to both use the cfgOption*() and poke data directly into the config struct. We can revisit it, but for now I think this works best.